examples/kokkos-tutorials/workload/bfs_bottomup.py

```diff
@@ -0,0 +1,164 @@
+from typing import Tuple
+
+import pykokkos as pk
+
+from parse_args import parse_args
+
+@pk.workload
+class Workload:
+    def __init__(self, N: int, M: int):
+        self.N: int = N
+        self.M: int = M
+
+        self.val: pk.View1D[pk.double] = pk.View([N*M], pk.double)
+        self.vis: pk.View1D[pk.double] = pk.View([N*M], pk.double)
+        self.Matrix: pk.View2D[pk.double] = pk.View([N, M], pk.double)
+        self.max_arr: pk.View1D[pk.double] = pk.View([N], pk.double)
+        self.max_arr2D: pk.View2D[pk.double] = pk.View([N, N], pk.double)
+
+        self.timer_result: float = 0
+        self.element: float = N*M
+
+
+        for i in range(N*M):
+            self.val[i] = N+M
+
+        for i in range(M):
+            self.vis[i] = 0
+
+
+        # Initialize the input matrix, can be design to be any binary matrix
+        # In this example, Matrix[0][1] & Matrix[0][3] will be 0, others will be 1
+        for j in range(N):
+            for i in range(M):
+                self.Matrix[j][i] = 1
+
+        self.Matrix[0][1] = 0
+        self.Matrix[0][3] = 0
+
+    @pk.main
+    def run(self):
+        timer = pk.Timer()
+        self.timer_result = timer.seconds()
+
+        # do the bfs
+        for i in range(self.N+self.M):
+            pk.parallel_for(self.element, self.check_vis)
+
+        # after bfs find maximum value in each row
+        pk.parallel_for("02", self.N, self.findmax)
```

Comment:
for the first argument in this call and calls below, you want to use/give unique names